src/test/java/com/amazon/dtasdk/utils/HttpUtilsTest.java

```diff
@@ -0,0 +1,45 @@
+package com.amazon.dtasdk.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+
+public class HttpUtilsTest {
+
+    @Test
+    public void testAppendUri() {
+        Assert.assertEquals("/", HttpUtils.appendUri("", ""));
+        Assert.assertEquals("basepath/two%2Bwords", HttpUtils.appendUri("basepath", "two+words"));
+        Assert.assertEquals("basepath/", HttpUtils.appendUri("basepath", null));
+        Assert.assertEquals("basepath/", HttpUtils.appendUri("basepath/", null));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath/", "test"));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath/", "/test"));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath", "/test"));
+        Assert.assertEquals("basepath/test//doubleslash", HttpUtils.appendUri("basepath", "/test//doubleslash"));
+        Assert.assertEquals("basepath/test/%2Fdoubleslash", HttpUtils.appendUri("basepath", "/test//doubleslash", true));
+    }
+
+    @Test
+    public void testIsUsingNonDefaultPort(){
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("http://test:80")));
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("https://test:443")));
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("scheme://other")));
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("scheme://other:0")));
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("scheme://other:-10")));
+
+        Assert.assertTrue(HttpUtils.isUsingNonDefaultPort(URI.create("http://test:9999")));
+        Assert.assertTrue(HttpUtils.isUsingNonDefaultPort(URI.create("http://test:443")));
+        Assert.assertTrue(HttpUtils.isUsingNonDefaultPort(URI.create("https://test:9999")));
+        Assert.assertTrue(HttpUtils.isUsingNonDefaultPort(URI.create("https://test:80")));
+    }
+
+    @Test
+    public void urlEncodeInputNullTrueOutputNotNull() {
```

Comment:
Can we run a test for some of these url replacements that contain two subsequent replaceable characters? For example: `"url+++++encode"`